src/format/opamFile.ml

```diff
@@ -3203,6 +3203,68 @@ module OPAMSyntax = struct
     in
     Pp.pp parse print
 
+  let rewrite_xfield = "x-allow-env-path-rewrite-on-windows"
+  let handle_env_paths =
+    let back_to_forward, forward_to_back =
+      let replace ~from ~to_ content =
+        let re =
+          Re.(compile @@
+              alt [
+                seq [ bos; char from ];
+                seq [ group (diff any (char '\\')); char from ]
+              ])
+        in
+        let f gr =
+          Printf.sprintf "%s%s"
+            (try Re.Group.get gr 1 with Not_found -> "") to_
+        in
+        Re.replace re ~f content
+      in
+      let b2f (_v,_op,content,_comment) =
+        _v, _op,
+        replace ~from:'/' ~to_:"\\" content,
+        _comment
+      in
+      let f2b (_v,_op,content,_comment) =
+        _v, _op,
+        replace ~from:'\\' ~to_:"/" content,
+        _comment
```

Comment:
Why not simply use the functions from `OpamSystem` that are already there?
```suggestion
      let b2f (v, op, content, comment) =
        (v, op, OpamSystem.back_to_forward content, comment)
      in
      let f2b (v, op, content, comment) =
        (v, op, OpamSystem.forward_to_back content, comment)
```

Reply:
The real answer is: i momentarily forget them.
Another answer is that those functions doesn't handle escaped slash or backslash. Maybe worth adding that too ?

Reply:
```suggestion
    let back_to_forward, forward_to_back =
      let replace ~from ~to_ content =
        let re =
          Re.(compile @@
              alt [
                seq [ bos; char from ];
                seq [ group (diff any (char '\\')); char from ]
              ])
        in
        let f gr =
          Printf.sprintf "%s%s"
            (try Re.Group.get gr 1 with Not_found -> "") to_
        in
        Re.replace re ~f content
      in
      let b2f (v, op, content, comment) =
        v, op,
        replace ~from:'/' ~to_:"\\" content,
        comment
      in
      let f2b (v, op, content, comment) =
        v, op,
        replace ~from:'\\' ~to_:"/" content,
        comment
```

Reply:
Hm here we are handling an already parsed string, so I don't think escapes should be an issue ?

Reply:
The idea is to be able to rewrite `first/still\/second/third` into `first\still/second\third`